src/main/webapp/dev/js/main/instructorCourseEnrollPage.js

```diff
@@ -47,34 +76,101 @@ const handsontable = new Handsontable(container, {
+    const spreadsheetData = enrollHandsontable.getData();
+    const dataPushToTextarea = getUpdatedHeaderString(enrollHandsontable.getColHeader());
     const userDataRows = getUserDataRows(spreadsheetData);
     $('#enrollstudents').text(userDataRows === '' ?
             '' : dataPushToTextarea + userDataRows); // only pushes header string if userDataRows is not empty
 }
 
+/**
+ * Loads existing student data into the spreadsheet interface.
+ * @param studentsData
+ * @param $panelHeading
+ * @param panelCollapse
+ * @param displayIcon
+ * @param toggleChevron
+ */
+function loadExistingStudentsData(studentsData, $panelHeading, panelCollapse, displayIcon, toggleChevron) {
+    dataHandsontable.loadData(getExistingStudentsData(
+            studentsData, dataHandsontable.getColHeader()));
+    toggleExistingStudentsPanel($panelHeading, panelCollapse,
+            displayIcon, toggleChevron);
+    dataHandsontable.render(); // needed as the view is buggy after collapsing the panel
+}
+
+/**
+ * Gets list of student data through an AJAX request.
+ * @param $panelHeading
+ * @param panelCollapse
+ * @param displayIcon
+ * @param toggleChevron
+ */
+/*  eslint no-unused-expressions: [2, { allowTernary: true }]   */
+function getAjaxStudentList($panelHeading, panelCollapse, displayIcon, toggleChevron) {
+    const $spreadsheetForm = $('#student-data-spreadsheet-form');
+
+    $.ajax({
+        type: 'POST',
+        url: '/page/instructorCourseEnrollAjaxPage',
+        cache: false,
+        data: {
+            courseid: $spreadsheetForm.children(`input[name="${ParamsNames.COURSE_ID}"]`).val(),
+            user: $spreadsheetForm.children(`input[name="${ParamsNames.USER_ID}"]`).val(),
+        },
+        beforeSend() {
+            displayIcon.html('<img height="25" width="25" src="/images/ajax-preload.gif">');
+        },
+        error() {
+            displayErrorExecutingAjax(displayIcon);
+        },
+        success(data) {
+            (data.students.length === 0) ? displayNoExistingStudents(displayIcon) :
+            loadExistingStudentsData(data.students, $panelHeading, panelCollapse, displayIcon, toggleChevron);
+        },
+    });
+}
+
+/**
+ * Function to collapse "Existing students" panel and loads existing students' data when the user clicks on it.
+ */
+function collapseExistingStudentsPanel() {
+    const $panelHeading = $(this);
+    const panelCollapse = $panelHeading.parent().children('.panel-collapse');
+    const displayIcon = $panelHeading.children('.display-icon');
+    const toggleChevron = $panelHeading.parent().find('.glyphicon-chevron-down, .glyphicon-chevron-up');
+
+    if ($panelHeading.attr('class').indexOf('ajax_submit') === -1) { // if panel is shown
+        hideExistingStudentsPanel($panelHeading, panelCollapse);
+        toggleChevronImage(panelCollapse, toggleChevron);
+    } else {
+        getAjaxStudentList($panelHeading, panelCollapse, displayIcon, toggleChevron);
```

Comment:
the naming is not very good here. The function doesn't just get the student list by AJAX but also load the data to the spreadsheet for display. In general, we should try to keep functions small and do only 1 thing. So you should keep the name. The fact that the data is further processed and loaded to the spreadsheet should be reflected here, or in another function. In case you need some technical help, there are a few ways to do this. The old way is to use a callback function. More modern javascript uses Promise and/or async/await functions.